hal/src/main/native/athena/SPI.cpp

```diff
@@ -218,7 +209,7 @@ void HAL_InitializeSPI(HAL_SPIPort port, int32_t* status) {
  */
 int32_t HAL_TransactionSPI(HAL_SPIPort port, uint8_t* dataToSend,
                            uint8_t* dataReceived, int32_t size) {
-  std::lock_guard<priority_recursive_mutex> sync(spiGetMutex(port));
+  std::lock_guard<std::mutex> sync(spiApiMutexes[port]);
```

Comment:
These will now throw an exception if port is out of range.  That is not what we want.  These always had an issue with out of range parameters, but we should go ahead and fix it.